spec/codecs/cef_spec.rb

```diff
@@ -360,369 +442,425 @@ def do_decode(codec, data)
-    it "should translate most known abbreviated CEF field names" do
-      decode_one(subject, translate_abbreviated_cef_fields) do |e|
-        validate(e)
-        insist { e.get("sourceAddress") } == "10.0.0.192"
-        insist { e.get("destinationAddress") } == "12.121.122.82"
-        insist { e.get("transportProtocol") } == "TCP"
-        insist { e.get("sourceHostName") } == "source.host.name"
-        insist { e.get("destinationHostName") } == "destination.host.name"
-        insist { e.get("sourcePort") } == "11024"
-        insist { e.get("destinationPort") } == "9200"
-        insist { e.get("eventOutcome") } == "Success"
-        insist { e.get("agentMacAddress")} == "00:80:48:1c:24:91"
+      let (:translate_abbreviated_cef_fields) {'CEF:0|security|threatmanager|1.0|100|trojan successfully stopped|10|src=10.0.0.192 dst=12.121.122.82 proto=TCP shost=source.host.name dhost=destination.host.name spt=11024 dpt=9200 outcome=Success amac=00:80:48:1c:24:91'}
+      it "should translate most known abbreviated CEF field names" do
+        decode_one(subject, translate_abbreviated_cef_fields) do |e|
+          validate(e)
+          insist { e.get(ecs_select[disabled:"sourceAddress",      v1:"[source][ip]"]) } == "10.0.0.192"
+          insist { e.get(ecs_select[disabled:"destinationAddress", v1:"[destination][ip]"]) } == "12.121.122.82"
+          insist { e.get(ecs_select[disabled:"transportProtocol",  v1:"[network][transport]"]) } == "TCP"
+          insist { e.get(ecs_select[disabled:"sourceHostName",     v1:"[source][domain]"]) } == "source.host.name"
+          insist { e.get(ecs_select[disabled:"destinationHostName",v1:"[destination][domain]"]) } == "destination.host.name"
+          insist { e.get(ecs_select[disabled:"sourcePort",         v1:"[source][port]"]) } == "11024"
+          insist { e.get(ecs_select[disabled:"destinationPort",    v1:"[destination][port]"]) } == "9200"
+          insist { e.get(ecs_select[disabled:"eventOutcome",       v1:"[event][outcome]"]) } == "Success"
+          insist { e.get(ecs_select[disabled:"agentMacAddress",    v1:"[agent][mac]"])} == "00:80:48:1c:24:91"
+        end
       end
-    end
 
-    let (:syslog) { "Syslogdate Sysloghost CEF:0|security|threatmanager|1.0|100|trojan successfully stopped|10|src=10.0.0.192 dst=12.121.122.82 spt=1232" }
-    it "Should detect headers before CEF starts" do
-      decode_one(subject, syslog) do |e|
-        validate(e)
-        insist { e.get('syslog') } == 'Syslogdate Sysloghost'
+      let (:syslog) { "Syslogdate Sysloghost CEF:0|security|threatmanager|1.0|100|trojan successfully stopped|10|src=10.0.0.192 dst=12.121.122.82 spt=1232" }
+      it "Should detect headers before CEF starts" do
+        decode_one(subject, syslog) do |e|
+          validate(e)
+          insist { e.get(ecs_select[disabled:'syslog',v1:'[log][syslog][header]']) } == 'Syslogdate Sysloghost'
+        end
       end
```

Comment:
@P1llus here is an example of a certain class of not-quite-legal CEF payloads that this codec handles, in which a syslog header of sorts is prepended to the actual CEF payload. Historically, we captured as a top-level `syslog`, but this proposes to capture as nested `log.syslog.header` instead. Do you have thoughts on the ideal place for this?

Reply:
Yeah from past experience CEF data does quite often have a syslog header added before the actual CEF data, it could be from special devices, wrong implementations or syslog forwarders.
Looking at the ECS spec for syslog, I do not believe there is a `log.syslog.header` field, but rather a `log.syslog` object, ref: https://www.elastic.co/guide/en/ecs/master/ecs-log.html#field-log-syslog.

Would you be able to confirm if or where you would like headers to be placed @ebeahan

Reply:
@yaauie Also if you want to test it with any test data, the 3 .log files here is raw CEF dat from various sources: https://github.com/elastic/beats/tree/master/x-pack/filebeat/module/cef/log/test

Reply:
There's currently not a designated field that is intended to capture the entirety of a syslog message header; perhaps a custom field?

The current fields under `log.syslog.*` are the syslog-related bits a header that wouldn't fit into an existing field elsewhere (e.g. `process.name`, `process.pid`, `host.hostname`, etc. could be present in the header)